### DISCUSSION TITLE
SFFS: Write Intermidiate Results to a file

### DISCUSSION THREAD

**Discussion by arilwan** [rasbt/mlxtend#1051] [Q&A] [answered]
As per the response to my previously requested feature #1043 , I understand to get Intermidiate results of the selector's attributes (feature evolution, scoring, etc) one has to wait for process to complete or issues a trigger a KeyboardInterrupt. In both cases, terminating the selection process.

Can you please guide me what section of the code should I change to continuously write the attributes values to a txt file that I can keep updating as the selection process proceeds?

Suppose those attributes are written to a file, I can do some anoalysis of the results during feature evolution (e.g. for the long  run example below that took over 2 weeks, and still running) `Features: 50/240, Features: 100/240, Features: 148/240` etc. without actually interrupting the running process.
```
[2023-06-14 08:01:25] Features: 149/240 -- score: 0.8947770129386831[Parallel(n_jobs=-1)]: Using backend LokyBackend with 8 concurrent workers.
[Parallel(n_jobs=-1)]: Done  25 tasks      | elapsed: 21.0min
[Parallel(n_jobs=-1)]: Done  91 out of  91 | elapsed: 60.4min finished
[Parallel(n_jobs=-1)]: Using backend LokyBackend with 8 concurrent workers.
[Parallel(n_jobs=-1)]: Done  25 tasks      | elapsed: 20.8min
[Parallel(n_jobs=-1)]: Done 149 out of 149 | elapsed: 100.3min finished
[Parallel(n_jobs=-1)]: Using backend LokyBackend with 8 concurrent workers.
[Parallel(n_jobs=-1)]: Done  25 tasks      | elapsed: 21.4min
[Parallel(n_jobs=-1)]: Done 148 out of 148 | elapsed: 89.4min finished

[2023-06-14 12:11:29] Features: 149/240 -- score: 0.8952890526770254[Parallel(n_jobs=-1)]: Using backend LokyBackend with 8 concurrent workers.
[Parallel(n_jobs=-1)]: Done  25 tasks      | elapsed: 18.2min
[Parallel(n_jobs=-1)]: Done  91 out of  91 | elapsed: 53.0min finished
[Parallel(n_jobs=-1)]: Using backend LokyBackend with 8 concurrent workers.
[Parallel(n_jobs=-1)]: Done  25 tasks      | elapsed: 18.5min
```
What section of the code should I change, to have these written to file?

**Comment by rasbt** [comment, marked as answer]
I think you could replace the `sys.stderr.write` lines with `f.write(...)`'s where `f` is a file object open for writing. Or, you could redirect the stderr out in your terminal to a file.

Then, you could could  insert `self.finalize_fit()` calls in the `fit` method; maybe here would be a good point: https://github.com/rasbt/mlxtend/blob/master/mlxtend/feature_selection/sequential_feature_selector.py#LL534C2-L536C40

**Comment by arilwan** [comment]
Thank you.
